Whiskey/Functions/Install-WhiskeyTool.ps1

```diff
@@ -286,12 +286,50 @@ function Install-WhiskeyTool
 
                         $TaskParameter[$ToolInfo.PathParameterName] = $nodePath
                     }
+                    'Dotnet'

```

Comment:
I'm trying to prevent us from falling into a monolithic function again, hence why I broke out all the logic into their own dedicated functions of `Resolve-WhiskeyDotNetSdkVersion`, `Install-WhiskeyDotNetSdk`, and `Set-WhiskeyDotNetGlobalJson`. Then `Install-WhiskeyTool` is the function that uses all those pieces to orchestrate the install into a state that is expected by the task requiring the tool.

Reply:
Is there a reason this other logic can't be put into `Install-WhiskeyDotNetSdk`?

Reply:
There is no technical reason I can't move it all into `Install-WhiskeyDotNetSdk`, only a design decision. My reasoning was creating/updating a `global.json` file should only happen in the context of a `DotNet` project's task, since its purpose is to pin a project to an SDK version. `Install-WhiskeyDotNetSdk` should not know/care of any Whiskey tasks or .NET Core project directories, it should only care about installing the requested version of the SDK. We can install the SDK anywhere but the `global.json` file should only be updated in either the working directory or build root. To have `Install-WhiskeyDotNetSdk` update the `global.json`, we'll either need to pass the build root and working directory to the function or give it an explicit path to check for and update the `global.json` file.

What do you think? Is that too strict of isolation between the functions?

Reply:
How about an `Install-WhiskeyDotNetTool` function?

Reply:
Yeah let's go with that. I'll replace the contents of the switch's `DotNet` code block with a `Install-WhiskeyDotNetTool` function.